tsdb/index/tsi1/index.go

```diff
@@ -575,11 +577,8 @@ func (i *Index) DropSeries(key []byte, ts int64) error {
 		fs := i.retainFileSet()
 		defer fs.Release()
 
-		// Check if that was the last series for the measurement in the entire index.
-		itr := fs.MeasurementSeriesIterator(mname)
-		if itr == nil {
-			return nil
-		} else if e := itr.Next(); e != nil {
+		mm := fs.Measurement(mname)
+		if mm == nil || mm.HasSeries() {
```

Comment:
nit: if a measurement is `nil`, then it can't have any series. Why not just make `HasSeries` check if the receiver is `nil` and return `false`? Then you don't need the `mm == nil || mm.HasSeries()`